### PR TITLE
add new API: LambdaDecay

### PR DESCRIPTION
**Pull request by zhwesky2010**
### PR types

New features
### PR changes

APIs
### Describe

Use the new base class `_LearningRateEpochDecay` to update the learning rate each epoch.

And add New API: 

`paddle.fluid.dygraph.LambdaDecay` 

A new learing rate strategy, support in dygraph mode.

### CODE REVIEW

File: python/paddle/fluid/dygraph/learning_rate_scheduler.py [hunk 1]
```diff
@@ -867,3 +866,95 @@ def _is_better(self, current, best):
...
+        else:
+            x = math.floor(\\frac{global\_step}{decay\_steps})
+
+        lr_lambda = lambda x: 0.95**x
+
+    Parameters:
+        learning_rate (Variable|float|int): The initial learning rate. It can be set to python float or int number.
+            If the type is Variable, it should be 1-D Tensor with shape [1], and the data type should be 'float32' 
+            or 'float64'.
+        decay_steps(int32): The decay step size. It determines the decay cycle.
```
Comment (lanxianghit) [inline python/paddle/fluid/dygraph/learning_rate_scheduler.py]: 只能是int32吗

Comment (zhwesky2010) [reply python/paddle/fluid/dygraph/learning_rate_scheduler.py]: python端的int就可以，历史的有两个写的是int32，其他都是int。统一改了文档描述

**Review by lanxianghit — DISMISSED** [review]

File: python/paddle/fluid/dygraph/learning_rate_scheduler.py [hunk 2]
```diff
@@ -1087,3 +1089,141 @@ def get_lr(self):
...
+                        loss = fluid.layers.reduce_mean(out)
+                        adam.minimize(loss)
+                    scheduler.epoch()
+
+                    print("epoch:%d, current lr is %f" % (epoch, adam.current_step_lr()))
+                    # epoch:0, current lr is 0.5        (eta_max) The beginning of 1th cycle
+                    # epoch:1, current lr is 0.452254
+                    # epoch:2, current lr is 0.327254
+                    # ...
+                    # epoch:5, current lr is 0          (eta_min) half of 1th cycle
```
Comment (phlrain) [inline python/paddle/fluid/dygraph/learning_rate_scheduler.py]: the current lr can be 0？

Comment (zhwesky2010) [reply python/paddle/fluid/dygraph/learning_rate_scheduler.py]: 现在按torch官网公式来了。epoch:5, current lr is 0.095492。而torch的框架代码计算出来第5个epoch会是0。

Comment (zhwesky2010) [reply python/paddle/fluid/dygraph/learning_rate_scheduler.py]: CosineAnnealing移到下个版本了，暂时不发。待向torch确认公式

**Review by lanxianghit — APPROVED** [review]

File: python/paddle/fluid/dygraph/learning_rate_scheduler.py (hunk 1 above)
Comment (zhwesky2010) [reply python/paddle/fluid/dygraph/learning_rate_scheduler.py]: Done，都改成int

**Review by swtkiwi — APPROVED** [review]
LGTM
